SVGPathSegValue::clone<>() rename to SVGPathSegValue::cloneInternal<>()

shallawa opened the pull request:
#### 9fc3aff2f3870364e33a99d82b67f2372472167f
<pre>
SVGPathSegValue::clone&lt;&gt;() rename to SVGPathSegValue::cloneInternal&lt;&gt;()
<a href="https://bugs.webkit.org/show_bug.cgi?id=254757">https://bugs.webkit.org/show_bug.cgi?id=254757</a>
rdar://107719787

Reviewed by Michael Catanzaro.

GCC 13 considers the template function SVGPathSegValue::clone&lt;&gt;() hides the virtual
function SVGPathSeg::clone&lt;&gt;() so the SVGPathSegValue super-classes can&apos;t override it.

Renaming it will make the classes in SVGPathSegValueImpl.h be complied on GCC 13
without the need for -Woverloaded-virtual.

* Source/WebCore/svg/SVGPathSeg.h:
* Source/WebCore/svg/SVGPathSegImpl.h:
* Source/WebCore/svg/SVGPathSegValue.h:
(WebCore::SVGPathSegValue::cloneInternal const):
(WebCore::SVGPathSegValue::clone const): Deleted.

Canonical link: <a href="https://commits.webkit.org/262690@main">https://commits.webkit.org/262690@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6c1d385dff8f10aa9674fd2dbec2ed56b7b4063

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2254 "Passed style check") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2285 "Build is in progress. Recent messages:") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2360 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3180 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2285 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2227 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/2371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2335 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2037 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2274 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/2371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2025 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/3038 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/2371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2006 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1892 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2084 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2052 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3200 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2066 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1992 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2014 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/561 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2187 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Ah, good idea. This looks like it should work.

shallawa committed:
SVGPathSegValue::clone<>() rename to SVGPathSegValue::cloneInternal<>()
https://bugs.webkit.org/show_bug.cgi?id=254757
rdar://107719787

Reviewed by Michael Catanzaro.

GCC 13 considers the template function SVGPathSegValue::clone<>() hides the virtual
function SVGPathSeg::clone<>() so the SVGPathSegValue super-classes can't override it.

Renaming it will make the classes in SVGPathSegValueImpl.h be complied on GCC 13
without the need for -Woverloaded-virtual.

* Source/WebCore/svg/SVGPathSeg.h:
* Source/WebCore/svg/SVGPathSegImpl.h:
* Source/WebCore/svg/SVGPathSegValue.h:
(WebCore::SVGPathSegValue::cloneInternal const):
(WebCore::SVGPathSegValue::clone const): Deleted.

Canonical link: https://commits.webkit.org/262690@main
